src/components/SelectNodeId.svelte

```diff
@@ -178,6 +150,91 @@
       filters.update(key, inp.value);
     };
   }
+
+  function _nodeValidator(value: number) {
+    value = +value;
+    if (typeof value !== "number") return "Please select a node.";
+    if (value < 1) return "Please select a valid node";
+  }
+
+  const nodeIdField: IFormField = {
+    label: "Node ID",
+    symbol: "nodeId",
+    type: "number",
+    placeholder: "Your Node ID",
+    validator: _nodeValidator,
+  };
+
+  interface IResources { cru: number; sru: number; hru: number; mru: number; ipv4u: number; } // prettier-ignore
+  interface ICapacity { total: IResources; used: IResources; } // prettier-ignore
+
+  let _ctrl: AbortController;
+  let _nodeId: number;
+  let validating: boolean = false;
+  $: {
+    if (profile && _nodeId !== data) {
+      if (!!_nodeValidator(data) && _ctrl) {
+        _ctrl.abort();
+        _ctrl = null;
+        validating = true;
+        status = null;
+      } else {
+        _nodeId = data;
+        if (_ctrl) _ctrl.abort();
+        _ctrl = new AbortController();
+
+        const { networkEnv } = profile;
+        const grid = new GridClient("" as any, "", "", null);
+        const { rmbProxy } = grid.getDefaultUrls(networkEnv as any);
+
+        validating = true;
+        status = null;
+        let _capacity: ICapacity;
+        fetch(`${rmbProxy}/nodes/${data}`, {
+          method: "GET",
+          signal: _ctrl.signal,
+        })
+          .then<{ capacity: ICapacity }>((res) => res.json())
+          .then(({ capacity }) => {
+            _capacity = capacity;
+            const { total, used } = capacity;
+            // prettier-ignore
+            let valid = (total.cru - used.cru) >= filters.cru &&
+                        ((total.sru - used.sru) / 1024 ** 3) >= filters.sru &&
+                        ((total.mru - used.mru) / 1024 ** 3) >= filters.mru;
+
+            if (!valid) {
+              status = "down";
```

Comment:
can we use "valid/invalid" status here? because I think that "up/down" does not represent the actual status.
please apply for the rest.

Reply:
### Updated